src/cli/subdoc_get.rs

```diff
@@ -0,0 +1,382 @@
+                    }
+                }
+                Ok(())
+            })?;
+        }
+    }
+
+    Ok(Value::List {
+        vals: results,
+        span: call.head,
+    }
+    .into_pipeline_data())
+}
+
+pub(crate) fn ids_from_input(
+    args: &Call,
+    input: PipelineData,
+    id_column: String,
+    ctrl_c: Arc<AtomicBool>,
+) -> Result<Vec<String>, ShellError> {
+    let mut ids: Vec<String> = input
+        .into_interruptible_iter(Some(ctrl_c))
+        .filter_map(move |v| match v {
+            Value::String { val, .. } => Some(val),
+            Value::Record { cols, vals, .. } => {
+                if let Some(idx) = cols.iter().position(|x| x.clone() == id_column) {
+                    if let Some(d) = vals.get(idx) {
+                        match d {
+                            Value::String { val, .. } => Some(val.clone()),
+                            _ => None,
+                        }
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        })
+        .collect();
+
+    if let Some(id) = args.positional_nth(1) {
+        if let Some(i) = id.as_string() {
+            ids.push(i);
+        }
+    }
+
+    Ok(ids)
+}
+
+#[derive(Debug)]
+struct GetResult {
+    error: Option<String>,
+    content: Option<Value>,
+    key: Option<String>,
+    cluster: String,
+    cas: Option<i64>,
+    id_column: Option<String>,
+}
+
+impl GetResult {
+    pub fn new(cluster: impl Into<String>) -> GetResult {
+        Self {
+            error: None,
+            content: None,
+            key: None,
+            cluster: cluster.into(),
+            cas: None,
+            id_column: None,
+        }
+    }
+
+    pub fn id_column(mut self, id_column: impl Into<String>) -> Self {
+        self.id_column = Some(id_column.into());
+        self
+    }
+
+    pub fn content(mut self, content: Value) -> GetResult {
+        self.content = Some(content);
+        self
+    }
+
+    pub fn key(mut self, key: String) -> GetResult {
+        self.key = Some(key);
+        self
+    }
+
+    pub fn cas(mut self, cas: i64) -> GetResult {
+        self.cas = Some(cas);
+        self
+    }
+
+    fn error(mut self, err: String) -> GetResult {
+        self.error = Some(err);
+        self
+    }
+
+    fn into_value(self, span: Span) -> Value {
+        let mut collected = NuValueMap::default();
+        collected.add_string(
+            self.id_column.unwrap_or_else(|| "id".to_string()),
+            self.key.unwrap_or_default(),
+            span,
+        );
+        collected.add("content", self.content.unwrap_or_default());
+        collected.add_i64("cas", self.cas.unwrap_or_default(), span);
+        collected.add_string("error", self.error.unwrap_or_default(), span);
+        collected.add_string("cluster", self.cluster, span);
+        collected.into_value(span)
+    }
+}
```

Comment:
I think all this already exists in doc_get?

Reply:
It is, have removed